src/fake_sign/modified_md5.rs

```diff
@@ -0,0 +1,250 @@
+// https://github.com/stainless-steel/md5/blob/3236fc93c26af909923767e65be87e45c47aab89/LICENSE.md
+//
+// Taken under the Apache 2.0 and MIT dual license to be included into this project with small
+// modifications.
+//
+// Apache 2.0
+//
+// Copyright 2015–2019 The md5 Developers
+//
+// Licensed under the Apache License, Version 2.0 (the “License”); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an “AS IS” BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+//
+// MIT:
+//
+// Copyright 2015–2019 The md5 Developers
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the “Software”), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#[derive(Clone)]
+pub struct Context {
+    buffer: [u8; 64],
+    count: [u32; 2],
+    pub state: [u32; 4], // hassle-rs modification
```

Comment:
Would it make sense to PR this back? Also, maybe a short overview of the crate mod(s?) at the top of the file in a comment would be nice :)

Reply:
I thought about it, but the crate didn't have any updates since 2019 and doesn't seem very active. However it is quite widely used. We can PR it back since this is the only real mod that we require. Did it like this for now because I don't want a git dependency here (since we can't push a release then).